From f06e6904ccb4d6ebb5b97b88b9e88fa11553856e Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@2eb42a87-3827-0410-b767-dc15055982e4>
Date: Thu, 12 Dec 2013 20:06:32 +0000
Subject: [PATCH 137/462] bugfix, extension Mantis 0029957: ConveyorComp:
 CUSS1.3 Conveyor (without ConveyorAEA)

AntProperties.xml
  TestDir changed to TestSrcDir
  added TestResourceDir
  added GenXMLSrcDir for generated source files from xml schema files (avoid mixing with other generated source files)

  added jars for SCXML:
    JarSCXML, JarLogging, JarJEXL, JarDigester, JarBeanUtils, JarBeanUtilsCore, JarXalan,
    JarApacheCollections change to JarCollections

AntTargets.xml
  fixed path element in target unit-test and compile-test-src use test.classpath and testsource.classpath
  use GenXMLSrcDir for generated source files from xml schema files
  use TestResourceDir
  TestDir changed to TestSrcDir


git-svn-id: svn://localhost/SelfServiceBuild/trunk@199 2eb42a87-3827-0410-b767-dc15055982e4
---
 SelfServiceBuild/Build/java/AntProperties.xml | 26 ++++++++++++++++---
 SelfServiceBuild/Build/java/AntTargets.xml    | 12 +++++----
 2 files changed, 29 insertions(+), 9 deletions(-)

diff --git a/SelfServiceBuild/Build/java/AntProperties.xml b/SelfServiceBuild/Build/java/AntProperties.xml
index 6e240fb..74747e3 100644
--- a/SelfServiceBuild/Build/java/AntProperties.xml
+++ b/SelfServiceBuild/Build/java/AntProperties.xml
@@ -86,9 +86,11 @@
 <property name="SrcDir"                     value="sources"/>
 <property name="ResourceDir"                value="resources"/>
 <property name="GenSrcDir"                  value="sources.generated"/>
+<property name="GenXMLSrcDir"               value="xmlsources.generated"/>
 <property name="ClassDir"                   value="ant_classes"/>
 <property name="TestClassDir"               value="ant_junit_classes"/>
-<property name="TestDir"                    value="test-sources"/>
+<property name="TestSrcDir"                 value="test-sources"/>
+<property name="TestResourceDir"            value="test-resources"/>
 
 <!-- ThirdParty-Software ========================================= -->
 
@@ -150,14 +152,14 @@
 <property name="JarMCClient"                value="${DIR_GLOBALS}/ThirdParty/mantisconnect/1.1.1.1/mantisconnect-client-api-1.1.1.1.jar"/>
 
 <!-- HTTPClient -->
-<property name="JarHTTPClient"                value="${DIR_GLOBALS}/ThirdParty/httpClient/4.2.3/lib/httpclient-4.2.3.jar"/>
+<property name="JarHTTPClient"              value="${DIR_GLOBALS}/ThirdParty/httpClient/4.2.3/lib/httpclient-4.2.3.jar"/>
 <property name="JarHTTPCore"                value="${DIR_GLOBALS}/ThirdParty/httpClient/4.2.3/lib/httpcore-4.2.2.jar"/>
 
 <!-- SVNKit -->
 <property name="JarSVNKit"                  value="${DIR_GLOBALS}/ThirdParty/svnkit/1.7.5/lib/svnkit-1.7.5-v1.jar"/>
 
 <!-- CLI -->
-<property name="JarCLI"                  value="${DIR_GLOBALS}/ThirdParty/commandlineInterface/1.2/commons-cli-1.2.jar"/>
+<property name="JarCLI"                     value="${DIR_GLOBALS}/ThirdParty/commandlineInterface/1.2/commons-cli-1.2.jar"/>
 
 <!-- Sonstiges -->
 <property name="JarCommonsConfiguration"    value="${DIR_GLOBALS}/ThirdParty/jakarta/commons/configuration/1.0/commons-configuration-1.0.jar"/>
@@ -171,13 +173,29 @@
 <property name="JarServlet"                 value="${DIR_GLOBALS}/ThirdParty/servlet-api/0.1/jar/servlet.jar"/>
 <property name="JarSNMP"                    value="${DIR_GLOBALS}/ThirdParty/snmp/snmp.jar"/>
 <property name="JarXerces"                  value="${DIR_GLOBALS}/ThirdParty/xerces-j/1.4.0/xerces-j.jar"/>
+
+<!-- XML-Beans -->
 <property name="JarXMLBeans"                value="${DIR_GLOBALS}/ThirdParty/XMLBeans/xmlbeans-2.5.0/lib/xbean.jar"/>
 <property name="JarJSR173"                  value="${DIR_GLOBALS}/ThirdParty/XMLBeans/xmlbeans-2.5.0/lib/jsr173_1.0_api.jar"/>
+
+<!-- velocity -->
 <property name="JarVelocity"                value="${DIR_GLOBALS}/ThirdParty/velocity/1.7/velocity-1.7.jar"/>
 <property name="JarVelocityTools"           value="${DIR_GLOBALS}/ThirdParty/velocity-tools/2.0/lib/velocity-tools-generic-2.0.jar"/>
-<property name="JarApacheCollections"       value="${DIR_GLOBALS}/ThirdParty/velocity/1.7/lib/commons-collections-3.2.1.jar"/>
+
+<!-- junit -->
 <property name="JarJunit"                   value="${DIR_GLOBALS}/ThirdParty/junit/4.10/junit-4.10.jar"/>
 
+<!-- scxml -->
+<property name="JarSCXML"                   value="${DIR_GLOBALS}/ThirdParty/commons-scxml/0.9/commons-scxml-0.9.jar"/>
+<property name="JarLogging"                 value="${DIR_GLOBALS}/ThirdParty/commons-logging/1.1.1/commons-logging-1.1.1.jar"/>
+<property name="JarJEXL"                    value="${DIR_GLOBALS}/ThirdParty/commons-jexl/1.1/commons-jexl-1.1.jar"/>
+<property name="JarDigester"                value="${DIR_GLOBALS}/ThirdParty/commons-digester/1.8/commons-digester-1.8.jar"/>
+<property name="JarCollections"             value="${DIR_GLOBALS}/ThirdParty/commons-collections/3.2.1/commons-collections-3.2.1.jar"/>
+<property name="JarBeanUtils"               value="${DIR_GLOBALS}/ThirdParty/commons-beanutils/1.8.3/commons-beanutils-1.8.3.jar"/>
+<property name="JarBeanUtilsCore"           value="${DIR_GLOBALS}/ThirdParty/commons-beanutils-core/1.7.0/commons-beanutils-core-1.7.0.jar"/>
+<property name="JarXalan"                   value="${DIR_GLOBALS}/ThirdParty/xalan/2.7.1/xalan-2.7.1.jar"/>
+
+
 
 <!-- JRE -->
 <property name="JarAppletPlugin"            value="${JAVA_HOME}/jre/lib/plugin.jar"/>
diff --git a/SelfServiceBuild/Build/java/AntTargets.xml b/SelfServiceBuild/Build/java/AntTargets.xml
index aebb0cf..376bd56 100644
--- a/SelfServiceBuild/Build/java/AntTargets.xml
+++ b/SelfServiceBuild/Build/java/AntTargets.xml
@@ -84,9 +84,10 @@
 
 <target name="create-xml-bean-file">
     <basename property="GenFile" file="${SchemaFile}" suffix=".xsd"/>    
+    <mkdir dir="${GenXMLSrcDir}/${GenFile}"/>
     <!-- Die angegebene Schema-Datei wird compiliert. -->
     <echo message="Compiling XML-Beans Schema: ${SchemaFile}"/>
-    <xmlbean schema="${SchemaFile}" srcgendir="${GenSrcDir}" destfile="${JarDir}/${GenFile}.jar" classpath="${JarXMLBeans};${JarJSR173}"/>
+    <xmlbean schema="${SchemaFile}" srcgendir="${GenXMLSrcDir}/${GenFile}" destfile="${JarDir}/${GenFile}.jar" classpath="${JarXMLBeans};${JarJSR173}"/>
 </target>
 
 <!-- create-xml-bean ================================================= -->
@@ -308,11 +309,11 @@
            verbose="${javac.verbose}"
            target="1.5">
         <classpath>
-          <pathelement path="${project.classpath}"/>
+          <path refid="testsource.classpath"/>
           <pathelement location="${JarName}"/>
           <pathelement location="${JarJunit}"/>
         </classpath>           
-        <src path="${TestDir}"/>
+        <src path="${TestSrcDir}"/>
     </javac>
 
 </target>
@@ -324,8 +325,9 @@
 
     <junit printsummary="yes" haltonfailure="yes">
       <classpath>
-        <pathelement path="${project.classpath}"/>
+        <path refid="test.classpath"/>
         <pathelement location="${TestClassDir}"/>        
+        <pathelement location="${TestResourceDir}"/>        
         <pathelement location="${JarName}"/>
         <pathelement location="${JarJunit}"/>        
       </classpath>
@@ -335,7 +337,7 @@
       <formatter type="xml"/>
 
       <batchtest fork="yes" todir="${ProjTestResultDir}">
-        <fileset dir="${TestDir}">
+        <fileset dir="${TestSrcDir}">
           <include name="**/Test*.java"/>
         </fileset>
       </batchtest>
-- 
2.41.0.windows.1

